### PR TITLE
HAL_ChibiOS: CubeOrange Enable Serial2 and Serial3 in bootloader hwdef

### LINKED ISSUE #17051: Bootloader: Serial Bootloader disabled on serial1 and serial2 on CubeOrange
## Bug report

**Issue details**
uploader.py fails when trying to load firmware through a real serial port to a CubeOrange. 

Test setup, CubeOrange connected to Jetson Xavier serial port at 921600 baud. Confirmed communication is working to mavproxy prior to try to load firmware
```
./uploader.py --port /dev/ttyTHS0 --baud-flightstack 921600 ardurover.apj 
Loaded firmware for 8c,0, size: 1441796 bytes, waiting for the bootloader...
If the board does not respond within 1-2 seconds, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
```

Cube seems to reboot, but firmware never starts uploading.

For reference same companion computer connected to a cube black:
```
./uploader.py --port /dev/ttyTHS0 --baud-flightstack 921600 ardurover-black.apj 
Loaded firmware for 9,0, size: 1416804 bytes, waiting for the bootloader...
If the board does not respond within 1-2 seconds, unplug and re-plug the USB connector.
Attempting reboot on /dev/ttyTHS0 with baudrate=921600...
If the board does not respond, unplug and re-plug the USB connector.
Found board 9,0 bootloader rev 5 on /dev/ttyTHS0
Bootloader Protocol: 5
OTP:
  type: ÿÿÿÿ
  idtype: =FF
  vid: ffffffff
  pid: ffffffff
  coa: //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////8=
  sn: 003f003f3435510738393730
ChipDes:
  family: STM32F42x
  revision: 3
Chip:
  20016419 STM32F42x_43x rev3 (no 1M flaw)
Info:
  flash size: 2080768
  board_type: 9 (fmuv3)
  board_rev: 0
Identification complete

Erase  : [====================] 100.0% (timeout: 7 seconds) 
Program: [====================] 100.0%
Verify : [====================] 100.0%
Rebooting.

```

**Version**
Master

**Platform**
[ x ] All

**Hardware type**
CubeOrange

### PR DESCRIPTION
**Pull request by jmachuca77**
In the CubeOrange bootloader serial ports 2 and 3 (telem1 and 2) are disabled so they cannot be used to upload firmware. This ports are enabled on the CubeBlack bootloader. This PR enables them for the CubeOrange.
Fixes #17051

### CODE REVIEW

**Comment by magicrub** [comment]
This looks good to me. It needs another commit of the newly generated bootloader binaries.

**Review by magicrub — APPROVED** [review]

**Comment by jmachuca77** [comment]
Added the binaries.

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat [hunk 1]
```diff
@@ -29,7 +29,21 @@ FLASH_BOOTLOADER_LOAD_KB 128
 define HAL_LED_ON 0
 
 # order of UARTs (and USB)
-SERIAL_ORDER OTG1 UART7
+SERIAL_ORDER OTG1 USART2 USART3 UART7
+
+# Another USART, this one for telem1. This one has RTS and CTS lines.
```
Comment (amilcarlucas) [inline libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat:34]: USART ??? shoudn't it be UART?

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat [hunk 2]
```diff
@@ -29,7 +29,21 @@ FLASH_BOOTLOADER_LOAD_KB 128
...
+# USART2 serial2 telem1
+PD3 USART2_CTS USART2
+PD4 USART2_RTS USART2
+PD5 USART2_TX USART2
+PD6 USART2_RX USART2
+
+# The telem2 USART, also with RTS/CTS available.
```
Comment (amilcarlucas) [inline libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat:41]: USART ??? shoudn't it be UART?

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
This requires approval by @bugobliterator 
There is a risk that this will result in some boards not booting if they generate a binary protocol on these uarts that causes the bootloader to disable its timeout and wait for a fw upload. Further improving the robustness of the bootloader protocol may be needed first

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat (hunk 1 above)
Comment (jmachuca77) [reply libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat:34]: Not according to the CubeOrange hwdef.dat file:
https://github.com/ArduPilot/ardupilot/blob/8544d00c97957ae2c2e28c0ceb15ca23e89d86b2/libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef.dat#L32

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat (hunk 2 above)
Comment (jmachuca77) [reply libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat:41]: https://github.com/ArduPilot/ardupilot/blob/8544d00c97957ae2c2e28c0ceb15ca23e89d86b2/libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef.dat#L142

**Comment by jmachuca77** [comment]
Any board that includes the file fmuv3/hwdef-bl.dat already has these ports enabled in the bootloader, 
https://github.com/ArduPilot/ardupilot/blob/8544d00c97957ae2c2e28c0ceb15ca23e89d86b2/libraries/AP_HAL_ChibiOS/hwdef/fmuv3/hwdef-bl.dat#L34
This includes CubeBlack, Pixhawk1, mRoX21, CubePurple, CubeBlack+, based on that and the number of those boards on the field without any reports of issues here, this change should be low risk.

**Comment by tridge** [comment]
I think we should defer this until we have added more robustness checks in the bootloader to prevent the zeroing of the timeout

**Comment by bugobliterator** [comment]
> I think we should defer this until we have added more robustness checks in the bootloader to prevent the zeroing of the timeout

will make a PR to supplement this.

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat [hunk 3]
```diff
@@ -29,7 +29,21 @@ FLASH_BOOTLOADER_LOAD_KB 128
 define HAL_LED_ON 0
 
 # order of UARTs (and USB)
-SERIAL_ORDER OTG1 UART7
+SERIAL_ORDER OTG1 USART2 USART3 UART7
+
+# Another USART, this one for telem1. This one has RTS and CTS lines.
+# USART2 serial2 telem1
+PD3 USART2_CTS USART2
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef-bl.dat:36]: better without RTS/CTS in bootloader

**Review by tridge — CHANGES_REQUESTED** [review]
needs rebase and rebuild of orange cube bootloader

**Comment by WickedShell** [comment]
FWIW I've had a lot of problems on my planes with the bootloader getting confused by either the telemetry radio or the GPS and hanging during attempted firmware updates on the plane because the data from one of those UART's is enough to get the port locked. I had the problem with a cube black constantly, and went to the effort to reflash the bootloader on all my systems to disable everything except the USB, as anything else lead to very reproducible problems with not being able to upload over USB once the vehicle was assembled. I'm probably in the minority though, and obviously I can keep flashing custom bootloaders, I just think it's worth noting that for a certain subset of users having more UART's enabled will lead to consistent failures to upload firmware.

**Comment by WickedShell** [comment]
As a further reference to my point: https://github.com/ArduPilot/ardupilot/issues/17452

**Comment by bugobliterator** [comment]
@WickedShell this PR https://github.com/ArduPilot/ardupilot/pull/17402 has been merged to improve the robustness.  Also the issue you mentioned is actually unrelated to your issue. That issue is seen even when there's nothing connected to UART line. My suspicion is that its a brownout issue, but will need further debugging to track it down.

**Comment by WickedShell** [comment]
I've had the brownout issue, but it definitely happens without browning
out. I have a solid power supply, and actually scooped it while it failed
once.

**Comment by tridge** [comment]
rebased, @bugobliterator please review

**Review by bugobliterator — APPROVED** [review]

**Pull request merged**